lib/ecto_ldap/adapter.ex

```diff
@@ -230,11 +177,81 @@ defmodule Ecto.Ldap.Adapter do
   def translate_value(%Ecto.Query.Tagged{value: value}), do: value
   def translate_value(other), do: other
 
-  def load(:id, value), do: {:ok, value}
+  def execute(_repo, query_metadata, prepared, params, preprocess, _options) do
+
+    search_response   = search(prepared, params)
+    fields            = ordered_fields(query_metadata.sources)
+    count             = count_fields(query_metadata.select.fields, query_metadata.sources)
+
+    {:ok, {:eldap_search_result, results, []}} = search_response
+
+    result_set = results
+    |> Enum.map(&create_rows(&1))
+    |> Enum.map(&prune_columns(&1, fields))
+    |> Enum.map(&generate_models(&1, preprocess, count))
```

Comment:
I may need a little more context -- I don't recall this block changing dramatically from what we had earlier, aside from some formatting and a slightly different field count.